File: CS/ClockTimeEditControl/ClockTimeEditInfoArgs.cs
using System.Drawing;
using DevExpress.XtraEditors.Controls;
using DevExpress.XtraEditors.ViewInfo;

namespace ClockTimeEditControl
{
    public class ClockTimeEditInfoArgs : VistaDateEditInfoArgs
    {
        public ClockTimeEditInfoArgs(DateEditCalendarBase calendar)
            : base(calendar)
        {
        }

        protected override void CalcClockInfo()
        {
            base.CalcClockInfo();

            ClockRect = new Rectangle(ClockRect.X - ClockRect.X, ClockRect.Y - ClockRect.Y, ClockRect.Width, ClockRect.Height);
            CalcClockPoints();
        }
    }
}

File: VB/ClockTimeEditControl/ClockTimeEditInfoArgs.vb
Imports Microsoft.VisualBasic
Imports System.Drawing
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.XtraEditors.ViewInfo

Namespace ClockTimeEditControl
	Public Class ClockTimeEditInfoArgs
		Inherits VistaDateEditInfoArgs
		Public Sub New(ByVal calendar As DateEditCalendarBase)
			MyBase.New(calendar)
		End Sub

		Protected Overrides Sub CalcClockInfo()
			MyBase.CalcClockInfo()

			ClockRect = New Rectangle(ClockRect.X - ClockRect.X, ClockRect.Y - ClockRect.Y, ClockRect.Width, ClockRect.Height)
			CalcClockPoints()
		End Sub
	End Class
End Namespace
